cmd/tg_bot/internal/environment.go

```diff
@@ -154,3 +163,137 @@ func (tgEnv *TelegramBotEnvironment) NodesListMessage(urls []string) (string, er
 	}
 	return w.String(), nil
 }
+
+func (tgEnv *TelegramBotEnvironment) SubscribeToAllAlerts() error {
+
+	for alertType, alertName := range entities.AlertTypes {
+		if tgEnv.IsAlreadySubscribed(alertType) {
+			return errors.Errorf("failed to subscribe to %s, already subscribed to it", alertName)
+		}
+		err := tgEnv.pubSubSocket.SetOption(mangos.OptionSubscribe, []byte{byte(alertType)})
+		if err != nil {
+			return err
+		}
+		tgEnv.mu.Lock()
+		tgEnv.subscriptions[alertType] = alertName
+		tgEnv.mu.Unlock()
+		log.Printf("Subscribed to %s", alertName)
+	}
+
+	return nil
+}
+
+func (tgEnv *TelegramBotEnvironment) SubscribeToAlert(alertType entities.AlertType) error {
+	alertName, ok := entities.AlertTypes[alertType] // check if such an alert exists
+	if !ok {
+		return errors.New("failed to subscribe to alert, unkown alert type")
+	}
+
+	if tgEnv.IsAlreadySubscribed(alertType) {
+		return errors.Errorf("failed to subscribe to %s, already subscribed to it", alertName)
+	}
+
+	err := tgEnv.pubSubSocket.SetOption(mangos.OptionSubscribe, []byte{byte(alertType)})
+	if err != nil {
+		return errors.Wrap(err, "failed to subscribe to alert")
+	}
+	tgEnv.mu.Lock()
+	tgEnv.subscriptions[alertType] = alertName
+	tgEnv.mu.Unlock()
+	log.Printf("Subscribed to %s", alertName)
+	return nil
+}
+
+func (tgEnv *TelegramBotEnvironment) UnsubscribeFromAlert(alertType entities.AlertType) error {
+	alertName, ok := entities.AlertTypes[alertType] // check if such an alert exists
+	if !ok {
+		return errors.New("failed to unsubscribe from alert, unkown alert type")
+	}
+
+	if !tgEnv.IsAlreadySubscribed(alertType) {
+		return errors.Errorf("failed to unsubscribe from %s, was not subscribed to it", alertName)
+	}
+
+	err := tgEnv.pubSubSocket.SetOption(mangos.OptionUnsubscribe, []byte{byte(alertType)})
+	if err != nil {
+		return errors.Wrap(err, "failed to unsubscribe from alert")
+	}
+	tgEnv.mu.RLock()
+	_, ok = tgEnv.subscriptions[alertType]
+	tgEnv.mu.RUnlock()
+	if !ok {
+		return errors.New("failed to unsubscribe from alert: was not subscribed to it")
+	}
+	tgEnv.mu.Lock()
+	delete(tgEnv.subscriptions, alertType)
+	tgEnv.mu.Unlock()
+	log.Printf("Unsubscribed from %s", alertName)
+	return nil
+}
+
+type Subscribed struct {
+	AlertName string
+}
+
+type Unsubscribed struct {
+	AlertName string
+}
+
+type Subscriptions struct {
+	SubscribedTo     []Subscribed
+	UnsubscribedFrom []Unsubscribed
+}
+
+func (tgEnv *TelegramBotEnvironment) SubscriptionsList() (string, error) {
+	tmpl, err := template.ParseFS(templateFiles, "templates/subscriptions.html")
+
+	if err != nil {
+		log.Printf("failed to construct a message, %v", err)
+		return "", err
+	}
+	var subcribedTo []Subscribed
+	tgEnv.mu.RLock()
+	for _, alertName := range tgEnv.subscriptions {
+		s := Subscribed{AlertName: alertName + "\n\n"}
+		subcribedTo = append(subcribedTo, s)
+	}
+	tgEnv.mu.RUnlock()
```

Comment:
Create special `MapR` method and append new element to the `subcribedTo` slice in closure. It can be helpful and guarantee that you don't forget to unlock mutex. And, btw, I think generics will be useful here.